MapForm.cs

```diff
@@ -490,106 +773,602 @@ void normalizePoint(ref Point a, ref Point b)
 					{
-						if (!chkWP[4 + wp].Checked) continue;
-						if (!_mapData[i].WPs[ord][wp].Enabled) continue;
+						if (!chkWP[4 + wp].Checked || !_mapData[i].WPs[ord][wp].Enabled) continue;
+						if (waypointInside(_mapData[i].WPs[ord][wp], ref m1, ref m2))
+						{
+							output.Add(new SelectionData(i, _mapData[i], ord, wp));
+						}
+					}
+				}
+			}
+		}
+
+		/// <summary>Returns true if two points are close enough to be considered a clicked point rather than a bounding box.</summary>
+		bool isPointClicked(ref Point p1, ref Point p2)
+		{
+			normalizePoint(ref p1, ref p2);
+			return (p2.X - p1.X <= 12 && p2.Y - p1.Y <= 12);
+		}
+
+		/// <summary>Determines if a waypoint (raw units) is within a bounding box formed by a top/left and bottom/right point.</summary>
+		bool waypointInside(Platform.BaseFlightGroup.BaseWaypoint wp, ref Point p1, ref Point p2)
+		{
+			int tx = wp.RawX;
+			int ty = wp.RawY;
+			int tz = wp.RawZ;
+			switch (_displayMode)
+			{
+				case Orientation.XY: return ((tx >= p1.X && tx <= p2.X) && (ty >= p1.Y && ty <= p2.Y));
+				case Orientation.XZ: return ((tx >= p1.X && tx <= p2.X) && (tz >= p1.Y && tz <= p2.Y));
+				case Orientation.YZ: return ((ty >= p1.X && ty <= p2.X) && (tz >= p1.Y && tz <= p2.Y));
+			}
+			return false;
+		}
+
+		/// <summary>Returns true if any specified objects are currently in our selection list.</summary>
+		bool isListObjectSelected(List<SelectionData> objects)
+		{
+			foreach (SelectionData test in objects)
+			{
+				foreach (SelectionData cur in _selectionList)
+				{
+					if (test.MapDataIndex == cur.MapDataIndex && test.WpIndex == cur.WpIndex && test.WpOrder == cur.WpOrder)
+						return true;
+				}
+			}
+			return false;
+		}
+		/// <summary>Returns true if the specified craft or any of its waypoints are selected.</summary>
+		bool isMapObjectSelected(int mapDataIndex)
+		{
+			for(int i = 0; i < _selectionList.Count; i++)
+				if (_selectionList[i].MapDataIndex == mapDataIndex)
+					return true;
+			return false;
+		}
+
+		/// <summary>Returns a size rating of the map object, based off its wireframe.</summary>
+		/// <remarks>Wireframes must be enabled. NOTE: This should probably be replaced with craft categories, but that would require additional changes, and more information in the craft data files.</remarks>
+		int getModelSizeRating(int mapDataIndex)
+		{
+			if (!_settings.WireframeEnabled)
+				return 0;
+			if (_platform == Settings.Platform.XWA && _mapData[mapDataIndex].Craft == 0x55)  // Hyper point
+				return 0;
+			WireframeInstance wire = _wireframeManager.GetOrCreateWireframeInstance(_mapData[mapDataIndex].Craft, _mapData[mapDataIndex].FgIndex);
+			if (wire == null || wire.ModelDef == null)
+				return 0;
+			int size = wire.ModelDef.LongestSpanMeters;
+			if (size > 1 && size < 40)
+				return 1;
+			else if (size >= 40 && size < 100)
+				return 2;
+			else if (size >= 100 && size < 450)
+				return 3;
+			else if (size >= 400)
```

Comment:
Should be 450 to match. Since we're re-using the same numbers in the checks, might be nice to have something like int size1 = 1, size2 = 40" etc so the lengths only exist once

Reply:
Good catch.  I tweaked the values at some point and must've overlooked that.  This was a lazy way of implementing some semblance of craft categories.  Ideally it would be entirely replaced, but category data would have to be compiled and added to the craft_data_xx.txt files for each platform.  A task for another time.